hawkular-alerts-rest/src/main/java/org/hawkular/alerts/rest/TriggersHandler.java

```diff
@@ -202,18 +305,107 @@ public Response updateTrigger(
+    public Response updateGroupTrigger(
+            @ApiParam(value = "Group Trigger id to be updated", required = true)
+            @PathParam("groupId")
+            final String groupId,
+            @ApiParam(value = "Updated group trigger definition", name = "trigger", required = true)
+            final Trigger groupTrigger) {
+        try {
+            if (groupTrigger != null && !isEmpty(groupId)) {
+                groupTrigger.setId(groupId);
+            }
+            definitions.updateGroupTrigger(tenantId, groupTrigger);
+            log.debugf("Trigger: %s ", groupTrigger);
+            return ResponseUtil.ok();
+
+        } catch (NotFoundException e) {
+            return ResponseUtil.notFound("Trigger " + groupId + " doesn't exist for update");
+
+        } catch (Exception e) {
+            log.debugf(e.getMessage(), e);
+            return ResponseUtil.internalError(e.getMessage());
+        }
+    }
+
+
+    @POST
+    @Path("/groups/members/{memberId}/orphan")
+    @Consumes(APPLICATION_JSON)
+    @ApiOperation(value = "Make a non-orphan member trigger into an orphan.")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Success, Trigger updated"),
+            @ApiResponse(code = 500, message = "Internal server error"),
+            @ApiResponse(code = 404, message = "Trigger doesn't exist/Invalid Parameters") })
+    public Response orphanMemberTrigger(
+            @ApiParam(value = "Member Trigger id to be made an orphan.", required = true)//
+            @PathParam("memberId")//
+            final String memberId) {
+        try {
+            Trigger child = definitions.orphanMemberTrigger(tenantId, memberId);
+            log.debugf("Orphan Member Trigger: %s ", child);
+            return ResponseUtil.ok();
+
+        } catch (NotFoundException e) {
+            return ResponseUtil.notFound("Member Trigger " + memberId + " doesn't exist for update");
+
+        } catch (Exception e) {
+            log.debugf(e.getMessage(), e);
+            return ResponseUtil.internalError(e.getMessage());
+        }
+    }
+
+    @POST
+    @Path("/groups/members/{memberId}/unorphan")
+    @Consumes(APPLICATION_JSON)
+    @ApiOperation(value = "Make a non-orphan member trigger into an orphan.")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Success, Trigger updated"),
+            @ApiResponse(code = 500, message = "Internal server error"),
+            @ApiResponse(code = 404, message = "Trigger doesn't exist/Invalid Parameters") })
+    public Response unorphanMemberTrigger(
+            @ApiParam(value = "Member Trigger id to be made an orphan.", required = true)//
+            @PathParam("memberId")//
+            final String memberId,
+            @ApiParam(required = true, name = "memberTrigger",
+                    value = "Only context and dataIdMap are used when changing back to a non-orphan.")//
+            final MemberTrigger memberTrigger) {
```

Comment:
If only context and dataIdMap are needed, I tend to a) use just an class that wraps these two arguments, b) use a generic Map and check that context and dataIdMap are needed.
Thinking in non-java REST clients, they would prefer to not load the full Trigger + Map of members, I think.
I didn't check it, but perhaps RESTEasy can force to have a full MemberTrigger deserialized and that can add a big json on client side.
On the other hand, if a client is reusing a json, then this is a minor issue, I think.